include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,767 @@
+                ++anchor_set_it;
+            }
+        }
+
+        //!\brief Construct from seqan3::gap_decorator_anchor_set and explicit position.
+        constexpr gap_decorator_anchor_set_iterator(gap_decorator_anchor_set const & host_,
+                                                    typename gap_decorator_anchor_set::size_type const pos_) noexcept :
+             host(&host_)
+        {
+            jump(pos_); // random access to pos
+        }
+        //!\}
+
+        /*!\name Arithmetic operators
+         * \{
+        */
+        //!\brief Pre-increment, returns updated iterator.
+        constexpr gap_decorator_anchor_set_iterator & operator++() noexcept
+        {
+            assert((host)); // host is set
+            ++pos;
+
+            if (pos < left_gap_end)
+            {   // we stay within the preceding gap stretch
+                // no op but must precede other cases
+            }
+            else if (anchor_set_it == (*host).anchors.end() || pos < (*anchor_set_it).first)
+            {   // proceed within the view since we are right of the previous gap but didn't arrive at the right gap yet
+                ++ungapped_view_pos;
+            }
+            else
+            {   // we arrived at the right gap and have to update the variables. ungapped_view_pos remains unchanged.
+                left_gap_end = (*anchor_set_it).first + (*anchor_set_it).second -
+                               ((anchor_set_it != (*host).anchors.begin()) ?
+                                    (*std::prev(anchor_set_it)).second : 0);
+                ++anchor_set_it;
+            }
+
+            return *this;
+        }
+
+        //!\brief Pre-decrement, returns updated iterator.
+        constexpr gap_decorator_anchor_set_iterator & operator--() noexcept
+        {
+            assert((host)); // host is set
+            --pos;
+
+            if (pos < left_gap_end)
+            {   // there was no gap before but we arrive at the left gap and have to update the variables.
+                if (anchor_set_it != (*host).anchors.begin())
+                {
+                    --anchor_set_it;
+                    if (anchor_set_it != (*host).anchors.begin())
+                    {
+                        auto prev = std::prev(anchor_set_it);
+                        left_gap_end = (*prev).first + (*prev).second -
+                                       ((prev != (*host).anchors.begin()) ? (*std::prev(prev)).second : 0);
+                    }
+                    else // [[unlikely]]
+                    {
+                        left_gap_end = 0;
+                    }
+                }
+            }
+            else if (anchor_set_it == (*host).anchors.end() || pos < (*anchor_set_it).first)
+            {   // we are neither at the left nor right gap
+                --ungapped_view_pos;
+            }
+            // else -> no op (we are still within the right gap stretch)
```

Comment:
How did we end up with so many branches in decrement when increment needs only three? (In the end, I would say it's preferable to have them in decrement, just wondering)

Reply:
acutally it has only two outer branches.. I can replace on branch by a ternary expression but thats it.